src/Traits/HasPermissions.php

```diff
@@ -152,14 +152,14 @@ public function filterPermission($permission, $guardName = null)
     {
         $permissionClass = $this->getPermissionClass();
 
-        if (is_string($permission)) {
+        if (is_string($permission) && ! PermissionRegistrar::isUid($permission)) {
             $permission = $permissionClass->findByName(
                 $permission,
                 $guardName ?? $this->getDefaultGuardName()
             );
         }
 
-        if (is_int($permission)) {
+        if (is_int($permission) || is_string($permission)) {
```

Comment:
Should is_string be isUid here?
And in HasRoles.

Reply:
Not because on line 155
```php 
if (is_string($permission) && ! PermissionRegistrar::isUid($permission)) {
```
So, we already check if is string and not uuid on `findByName`, if it still is string means that it is an uuid for `findById`